packages/devextreme/testing/testcafe/tests/scheduler/layout/templates/appointmentTemplate.ts

```diff
@@ -66,3 +76,274 @@ fixture.disablePageReloads`Layout:Templates:appointmentTemplate`
     });
   });
 });
+
+const viewTypes: ViewType[] = [
+  'agenda',
+  'day',
+  'week',
+  'workWeek',
+  'month',
+  'timelineDay',
+  'timelineWeek',
+  'timelineWorkWeek',
+  'timelineMonth',
+];
+
+const groupOrientations: Orientation[] = ['horizontal', 'vertical'];
+const scrollModes: ScrollMode[] = ['standard', 'virtual'];
+const rtlEnabledOptions: boolean[] = [false, true];
+
+const testOptions = generateOptionMatrix({
+  viewType: viewTypes,
+  groupOrientation: groupOrientations,
+  scrollMode: scrollModes,
+  rtlEnabled: rtlEnabledOptions,
+}, [
+  // Not supported
+  {
+    viewType: 'agenda',
+    scrollMode: 'virtual',
+  },
+  // Not supported
+  {
+    viewType: 'agenda',
+    groupOrientation: 'horizontal',
+  },
+  {
+    viewType: 'day',
+    groupOrientation: 'vertical',
+    scrollMode: 'standard',
+  },
+  {
+    viewType: 'week',
+    groupOrientation: 'vertical',
+    scrollMode: 'standard',
+  },
+  {
+    viewType: 'workWeek',
+    groupOrientation: 'vertical',
+    scrollMode: 'standard',
+  },
+  {
+    viewType: 'day',
+    groupOrientation: 'horizontal',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'week',
+    groupOrientation: 'horizontal',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'workWeek',
+    groupOrientation: 'horizontal',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'month',
+    groupOrientation: 'horizontal',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'timelineDay',
+    groupOrientation: 'vertical',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'timelineWeek',
+    groupOrientation: 'vertical',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'timelineWorkWeek',
+    groupOrientation: 'vertical',
+    rtlEnabled: true,
+  },
+  {
+    viewType: 'timelineMonth',
+    groupOrientation: 'vertical',
+    rtlEnabled: true,
+  },
+]);
+
+testOptions.forEach(({
+  viewType,
+  groupOrientation,
+  scrollMode,
+  rtlEnabled,
+}) => {
+  const currentDate = new Date(2024, 0, 2);
+  const HOUR = 1000 * 60 * 60;
+
+  const startDayHour = 8;
+  const endDayHour = 20;
+
+  let resourceCount = 30;
```

Comment:
Let's move the logic of setting the ```resourceCount``` value into a separate function.
It makes the code more linear and obvious.
Now, it takes time to see all cases os setting ```resourceCount``` variable :)

E.g.
```typescript
  // some variables
 const getResourceCount = (
   viewType: ViewType,
   scrollMode: ScrollMode,
   orientation: Orientation,
 ) => {
    switch (true) {
      case viewType == 'week' && scrollMode == 'virtual' && orientation == 'horizontal':
        return 10;
      // other cases
     default:
       return 30;
  }
  
  // other code
  
  testOptions.forEach({...}) => {
  // other code
  
  const resourceCount = getResourceCount(viewType, scrollMode, orientation);
  
  // other code
  }
```